### PR TITLE
Initialize pkg/pub_dartdoc along the global dartdoc intialization.

### PR DESCRIPTION
**Pull request by isoos**
I'd like to keep the two versions side-by-side and test it before removing the global one.

### CODE REVIEW

File: app/lib/shared/service_utils.dart [hunk 1]
```diff
@@ -272,6 +278,17 @@ Future initDartdoc(Logger logger) async {
   }
 }
 
+Future initPkgDartdoc(Logger logger) async {
+  logger.info('Initializing pkg/pub_dartdoc');
+  final dir = Platform.script.resolve('../../pkg/pub_dartdoc').toFilePath();
+  final pr = await Process.run('pub', ['get'], workingDirectory: dir);
```
Comment (sortie) [inline app/lib/shared/service_utils.dart]: Perhaps have a time out if the command doesn't halt.

**Comment by isoos** [comment]
> What exactly is happening here?

Instead of using the SDK's `dartdoc` (or the globally activated dartdoc), we want to use `pkg/pub_dartdoc`, which extends it with further data extraction (for search and scoring). We could depend on that project as a regular library, but that would tie us to the dependencies of `dartdoc`, and we could be on a different upgrade schedule. Instead, we run it as it were a separate app, and for that, we need to fetch the dependencies before calling it.

File: app/lib/shared/service_utils.dart (hunk 1 above)
Comment (isoos) [reply app/lib/shared/service_utils.dart]: Good point, adding it.

**Comment by isoos** [comment]
Looks good in staging, merging.

**Pull request merged**